versioned/storage/store/src/main/java/org/projectnessie/versioned/storage/versionstore/VersionStoreImpl.java

```diff
@@ -859,12 +864,15 @@ public PaginationIterator<Diff> getDiffs(
 
     ContentMapping contentMapping = new ContentMapping(persist);
 
-    Predicate<ContentKey> contentKeyPredicate = keyRestrictions.contentKeyPredicate();
+    BiPredicate<ContentKey, Content.Type> contentKeyPredicate =
+        keyRestrictions.contentKeyPredicate();
     Predicate<DiffEntry> keyPred =
         contentKeyPredicate != null
             ? d -> {
               ContentKey key = storeKeyToKey(d.key());
-              return key != null && contentKeyPredicate.test(key);
+              return key != null
```

Comment:
This looks better imho, and corresponds to what the old model does with `KeyFilterPredicate` (it's applied to both "from" and "to").